source/common/router/config_impl.cc

```diff
@@ -525,6 +526,15 @@ RouteEntryImplBase::RouteEntryImplBase(const CommonVirtualHostSharedPtr& vhost,
     direct_response_body_provider_ = std::move(provider_or_error.value());
   }
 
+  if (route.has_direct_response() && route.direct_response().has_body_format()) {
```

Comment:
FYI in the C++ implementation of protobuf you don't have to check two things for a layered presence check, you can just check `route.direct_response().has_body_format()`, because direct_response() will return a reference to the default uninitialized message, and that won't have a body_format, so the result will be the same.